### PR TITLE
feat: Added frontend for newsletter section ( exploring backend for the same )

### PR DESCRIPTION
**Pull request by Adeesh-bode**
## Related Issue

Issue Number :  #295 
Part of GirlScript summer of code
Added frontent  for newletter section

## Screenshots
Attached
Screenshot (114)

## Summary by CodeRabbit

- **New Features**
  - Introduced a `Newsletter` component for email sign-ups and social media sharing.
  - Added an optional `type` property to buttons, allowing for different button types (button, submit, reset).

- **Style**
  - Updated global styles to include a red outline.

### CODE REVIEW

File: src/components/core/newsletter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"use client";
+import React from "react";
+import { Linkedin, Facebook ,Twitter , Instagram  } from 'lucide-react';
+
+const Newsletter: React.FC = () => {
+    return <div className="w-full ">
+        <div className="flex flex-col justify-center items-center  gap-6 px-20 py-10 ">
+            <h4 className="text-xs uppercase">Newsletter</h4>
```
Comment (yashd-dev) [inline src/components/core/newsletter/index.tsx:8]: The font size is way too small

File: src/components/core/newsletter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            <h2  className="text-xs md:text-base uppercase text-primary-light text-center">Sign up for latest updates and offers</h2>
+            <div className="flex flex-col gap-4 md:flex-row">
```
Comment (yashd-dev) [inline src/components/core/newsletter/index.tsx:10]: In light mode, it is unclear if an input is even present, put something like a border around it for light mode.

**Comment by yashd-dev** [comment]
I feel that the backend also has to be included in this PR to merge this into main. We can't ship incomplete features, @Sanchitbajaj02 thoughts?

**Comment by Sanchitbajaj02** [comment]
> I feel that the backend also has to be included in this PR to merge this into main. We can't ship incomplete features, @Sanchitbajaj02 thoughts?

Yes, backend is must. Also, @Adeesh-bode there is no need to create new button, I have already configured a customized button with an action, please use that.

**Comment by Sanchitbajaj02** [comment]
@Adeesh-bode any updates on this PR?

**Comment by Adeesh-bode** [comment]
I thought you rejected my work so I wasn't working on it... is it possible if only frontend part is done by me ( I would explore app write and do the  if not possible ). I am asking cause a person familiar with app write can do this job better.

**Comment by Sanchitbajaj02** [comment]
> I thought you rejected my work so I wasn't working on it... is it possible if only frontend part is done by me ( I would explore app write and do the if not possible ). I am asking cause a person familiar with app write can do this job better.

I never mentioned the rejection. If you scroll up we suggested the changes that needs to be done to get it merged.

Also it is alright if you don't know or not familiar with Appwrite. The PR will be merged, just do the changes that are suggested above

**Comment by Adeesh-bode** [comment]
ok I will do the suggested changes and raise pr by day end..

**Comment by Adeesh-bode** [comment]
hey sanchit , I started working on using already made custom button , which is used for redirection by href but the subscribe button is not redirecting it is triggering and onSubmit action for backend. I will still want to use your made custom button than I would have to make states which will pass from child component to parent i.e from custom button component to newsletter component ( which is not recommend ) or will have to use redux global state . If you say so I will work on it, as I looked in  other part of project like login & register page where such trigger buttons are used , they also didn't used custom button.

**Comment by Adeesh-bode** [comment]
other than button suggestion other frontend suggestion are considered.

**Comment by Sanchitbajaj02** [comment]
> hey sanchit , I started working on using already made custom button , which is used for redirection by href but the subscribe button is not redirecting it is triggering and onSubmit action for backend. I will still want to use your made custom button than I would have to make states which will pass from child component to parent i.e from custom button component to newsletter component ( which is not recommend ) or will have to use redux global state . If you say so I will work on it, as I looked in other part of project like login & register page where such trigger buttons are used , they also didn't used custom button.

For the button, you can introduce an option on click event field. Like either the href will work or the onclick event will work on that button. This will help streamline the button component throughout the application

**Comment by Adeesh-bode** [comment]
done, utilized the prebuilt ui component for button

**Comment by hvmanker** [comment]
can you send a. video of it also is the backend also ready for the same mentioned above also

**Comment by Adeesh-bode** [comment]
I didn't know app write backend so I asked sanchit , that can I do the frontend. So I did only frontend part. The image for frontend is attached in above message

**Pull request merged**